crates/jstz_api/src/stream/queuing_strategy/builtin.rs

```diff
@@ -0,0 +1,216 @@
+use crate::{
+    idl,
+    stream::{
+        queuing_strategy::size::{
+            ByteLengthQueuingStrategySizeAlgorithm, CountQueuingStrategySizeAlgorithm,
+        },
+        tmp::get_jsobject_property,
+    },
+};
+use boa_engine::{
+    js_string, object::Object, property::Attribute, value::TryFromJs, Context, JsArgs,
+    JsNativeError, JsResult, JsValue, NativeFunction,
+};
+use boa_gc::{Finalize, GcRefMut, Trace};
+use jstz_core::{
+    accessor,
+    js_fn::JsCallableWithoutThis,
+    native::{Accessor, ClassBuilder, JsNativeObject, NativeClass},
+};
+
+/// [Streams Standard - § 7.1.][https://streams.spec.whatwg.org/#qs-api]
+/// > ```
+/// > dictionary QueuingStrategyInit {
+/// >   required unrestricted double highWaterMark;
+/// > };
+/// > ```
+pub struct QueuingStrategyInit {
+    pub high_water_mark: idl::UnrestrictedDouble,
+}
+
+impl TryFromJs for QueuingStrategyInit {
+    fn try_from_js(value: &JsValue, context: &mut Context<'_>) -> JsResult<Self> {
+        let this = value.to_object(context)?;
+        let high_water_mark: idl::UnrestrictedDouble =
+            get_jsobject_property(&this, "highWaterMark", context)?
+                .try_js_into::<Option<idl::UnrestrictedDouble>>(context)?
+                .ok_or_else(|| {
+                    JsNativeError::typ()
+                        .with_message("Missing required highWaterMark property")
+                })?;
```

Comment:
Hm. The whole `impl TryFromJs for QueuingStrategyInit` should be derived. The `get_jsobject_property` function is just something I added as a temporary fix because the derived `TryFromJs` didn't work for `UnderlyingSource`, and it seems to also not work for `QueuingStrategyInit`: running `strat = new CountQueuingStrategy({highWaterMark: 1})` yields an error `✕  ERROR  Uncaught TypeError: cannot get property high_water_mark of value` if I use
```
#[derive(TryFromJs)]
pub struct QueuingStrategyInit {
    pub high_water_mark: idl::UnrestrictedDouble,
}
```
(which may be caused by the difference between Rust naming convensions and JS ones, i.e. `high_water_mark` vs `highWaterMark`, but I haven't looked into it)

Reply:
See https://tezos-dev.slack.com/archives/C061SSDBN69/p1701430478351919